lib/Rides.dart

```diff
@@ -0,0 +1,198 @@
+    if (response.statusCode == 200) {
+      String responseBody = response.body;
+      List<Ride> rides = _ridesFromJson(responseBody);
+      Ride currentRide;
+      if(rides.length > 0) {
+        currentRide = rides[0];
+        rides.removeAt(0);
+      }
+      var d = _RideData(rides,currentRide);
+      return d;
+    } else {
+      throw Exception('Failed to load rides.');
+    }
+    */
+  }
+
+  List<Ride> _ridesFromJson(String json) {
+    var data = jsonDecode(json)["data"];
+    List<Ride> res = data
+        .map<Ride>((e) => Ride.fromJson(e))
+        .where((Ride e) => e.driverId.contains(widget.driverId))
+        .toList();
+    res.sort((a, b) => a.startTime.compareTo(b.startTime));
+    return res;
+  }
+
+  // TODO: replace this when name is stored somewhere
+  String getNameFromSharedStateSomewhere() {
+    return "Chris";
+  }
+
+  Widget _futureRide(BuildContext context, _RideData d, int index) {
+    return FutureRide(d.rides[index]);
+  }
+
+  Widget _emptyPage(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        SizedBox(height: 195),
+        Center(
+            child: Column(
+          children: <Widget>[
+            Image(
+              image: AssetImage('assets/images/steeringWheel@3x.png'),
+              width: MediaQuery.of(context).size.width * 0.2,
+              height: MediaQuery.of(context).size.width * 0.2,
+            ),
+            SizedBox(height: 22),
+            Text(
+              'Congratulations! You are done for the day. \n'
+              'Come back tomorrow!',
+              textAlign: TextAlign.center,
+            )
+          ],
+        )),
+      ],
+    );
+  }
+
+  Widget _mainPage(BuildContext context, _RideData data) {
+    return Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        // height: 20,
+        children: <Widget>[
+          LeftSubheading(heading: 'Upcoming Ride'),
+          Center(
+            child: CurrentRide(data.currentRide),
+          ),
+          SizedBox(height: 16.0),
+          LeftSubheading(heading: 'Today\'s Schedule'),
+          Expanded(
+            child: ListView.separated(
+              itemCount: data.rides.length,
+              itemBuilder: (BuildContext c, int index) =>
+                  _futureRide(c, data, index),
+              separatorBuilder: (BuildContext context, int index) => Divider(),
+              padding: EdgeInsets.only(left: 24.0, right: 24.0, bottom: 5.0),
+              shrinkWrap: true,
+            ),
+          )
+        ]);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          Greeting(getNameFromSharedStateSomewhere()),
+          Expanded(
+              child: FutureBuilder<_RideData>(
+                  future: rideData,
```

Comment:
I think you can make the future the fetchRides function itself. Setting the ride data to that in the initState function is unnecessary, I believe.